tests/unit/s2n_tls13_handshake_test.c

```diff
@@ -412,20 +412,22 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_set_cipher_preferences(server_conn, "default_tls13"));
 
         /* Client sends ClientHello */
+        EXPECT_EQUAL(s2n_conn_get_current_message_type(client_conn), CLIENT_HELLO);
```

Comment:
the previous tests are asserting that states after s2n_handshake_write_io() is called, should those tests be kept in addition to the "states before" assertions?

Reply:
I don't think they are necessary because the coverage hasn't changed for s2n_handshake_io.c, and every line in the `s2n_handshake_write_io` is covered multiple times. I would like to improve on functional and integration tests though. I think we are lacking in that regard.